utils/data.py

```diff
@@ -364,37 +377,66 @@ def fload_and_patch(self, selectedPatients, selectedDatasets):
+
 
                 if os.path.exists(currentDataDir):
-                    # get list with all paths of dicoms for current patient and current dataset
+                    # get list with all paths of dicoms for current patient and current dataset- Magnitude data
                     fileNames = os.listdir(currentDataDir)
                     fileNames = [os.path.join(currentDataDir, f) for f in fileNames]
 
+                if os.path.exists(currentDataDirphase):
+                    # get list with all paths of dicoms for current patient and current dataset - Phase data
+                    fileNamesphase = os.listdir(currentDataDirphase)
+                    fileNamesphase = [os.path.join(currentDataDirphase, f) for f in fileNamesphase]
+
                     # read DICOMS
                     dicomDataset = [pydicom.read_file(f) for f in fileNames]
-                    # TODO: add here reading in of phase images
+                    dicomDatasetphase = [pydicom.read_file(f) for f in fileNamesphase]
+                    # TODO: add here reading in of phase images -Done
 
-                    # Combine DICOM Slices to a single 3D image (voxel)
+                    # Combine DICOM Slices to a single 3D image (voxel)-Magnitude part
                     try:
                         voxel_ndarray, _ = dicom_np.combine_slices(dicomDataset)
                         voxel_ndarray = voxel_ndarray.astype(float)
                         voxel_ndarray = np.swapaxes(voxel_ndarray, 0, 1)
                     except dicom_np.DicomImportException as e:
                         # invalid DICOM data
                         raise
+                    # Combine DICOM Slices to a single 3D image (voxel)-Phase part
+                    try:
+                        voxel_ndarray_phase, ijk_to_xyz_phase = dicom_np.combine_slices(dicomDatasetphase)
+                        voxel_ndarray_phase = voxel_ndarray_phase.astype(float)
+                        voxel_ndarray_phase = np.swapaxes(voxel_ndarray_phase, 0, 1)
+                    except dicom_np.DicomImportException as e:
+                        # invalid DICOM data
+                        raise
 
-                    # normalization of DICOM voxel
+                    # Clean voxel phase - To remove the background noise in the phase part
+                    binMask = voxel_ndarray > 0.05
+                    voxel_ndarray_phase = np.multiply(voxel_ndarray_phase, binMask)
+
+                    # normalization of DICOM voxel-Magnitude and phase part
                     rangeNorm = [0, 1]
                     norm_voxel_ndarray = (voxel_ndarray - np.min(voxel_ndarray)) * (rangeNorm[1] - rangeNorm[0]) / (
                             np.max(voxel_ndarray) - np.min(voxel_ndarray))
+                    norm_voxel_ndarray_phase = (voxel_ndarray_phase - np.min(voxel_ndarray_phase)) * (
```

Comment:
phase should either be scaled into [-1, 1] (representing -pi to pi)
or to [0, 2*pi]
--> obviously this decision impacts the building of the architecture, especially the input layer